Add remote store main page updates. Add shallow snapshots (#5078)
opensearch-project/documentation-website: _tuning-your-cluster/availability-and-recovery/remote-store/snapshot-interoperability.md

---
layout: default
title: Shallow snapshots
nav_order: 15
parent: Remote-backed storage
grand_parent: Availability and recovery
---

# Shallow snapshots

> Shallow copy snapshots allow you to reference data from an entire remote-backed segment instead of storing all of the data from the segment in a snapshot. This makes accessing segment data faster than normal snapshots, because segment data is not stored in the snapshot repository.

Reviewer: ```suggestion
Shallow copy snapshots allow you to reference data from an entire remote-backed segment instead of storing all of the data from the segment in a snapshot. This makes accessing segment data faster than normal snapshots because segment data is not stored in the snapshot repository.
```

> Shallow copy snapshots allow you to reference data from an entire remote-backed segment instead of storing all of the data from the segment in a snapshot. This makes accessing segment data faster than normal snapshots, because segment data is not stored in the snapshot repository.

Reviewer: In the last sentence, I would either remove "than normal snapshots" or change to "than when using normal snapshots".

> Shallow copy snapshots allow you to reference data from an entire remote-backed segment instead of storing all of the data from the segment in a snapshot. This makes accessing segment data faster than using normal snapshots because segment data is not stored in the snapshot repository.

Reviewer 2: Shallow copy snapshots allows you to reference data directly from remote store instead of storing all of the segment data again in snapshot repository. These snapshots gets created faster than normal snapshots because segment data is not stored in the snapshot repository.

## Enabling shallow snapshots

Use the [Cluster Settings API]({{site.url}}{{site.baseurl}}/api-reference/cluster-api/cluster-settings/) to enable the `remote_store_index_shallow_copy` repository setting, as shown in the following example:

> ```bash

Reviewer 2: @Naarcha-AWS we don't need to update cluster settings api, we need to call the PUT _snapshot/<repository> api to enable this

Author: Could you post the exact call with request body here?

Reviewer 2: something like this:
```
curl -X PUT "localhost:9200/_snapshot/snap_repo?pretty" -H 'Content-Type: application/json' -d'
{
        "type": "s3",
        "settings": {
            "bucket": "test-bucket",
            "base_path": "daily-snaps",
            "remote_store_index_shallow_copy": true
        }
    }
'
```

PUT _cluster/settings
{
   "persistent":{
      "remote_store_index_shallow_copy": true
   }
}
```
{% include copy-curl.html %}

Once enabled, all requests using the [Snapshot API]({{site.url}}{{site.baseurl}}/api-reference/snapshots/index/) will remain the same for all snapshots. After the setting is enabled, we recommend not disabling the setting. Doing so could affect data durability. 

## Considerations

Consider the following before using shallow copy snapshots:

> - Shallow copy snapshots only work for remote-backed indexes.

Reviewer 2: indexes -> indices

- All nodes in the cluster must use OpenSearch 2.10 or later to take advantage of shallow copy snapshots.

> - The file size difference of shallow copy snapshot shards is zero, because no segment data is stored in the snapshot itself.

Reviewer 3: Could this be something like ...
```suggestion
- There is no difference in file size between standard (regular, normal, primary or replica???) shards and shallow copy snapshot shards because no segment data is stored in the snapshot itself.
```

> - There is no difference in file size between standard shards and shallow copy snapshot shards because no segment data is stored in the snapshot itself.

Reviewer 2: Also, this line need to be updated. so, we show the incremental file count and size in bytes between the last snapshot and current snapshot in snapshot status API (https://opensearch.org/docs/latest/api-reference/snapshots/get-snapshot-status/#snapshot-file-stats). in case of shallow copy snapshot incremental `file count` and `size in bytes` will be zero.

> - There is no difference in file size between standard (regular, normal, primary or replica???) shards and shallow copy snapshot shards because no segment data is stored in the snapshot itself.

Reviewer: I'm assuming either the entire parenthetical or the question marks within it should be removed?

Author: This was a suggestion from Chris I misunderstood. Adjusting.

> - There is no difference in file size between standard (regular, normal, primary or replica???) shards and shallow copy snapshot shards because no segment data is stored in the snapshot itself.

Author: ```suggestion
- There is no difference in file size between standard shards and shallow copy snapshot shards because no segment data is stored in the snapshot itself.
```

> - Searchable snapshot are not supported inside shallow copy snapshots.

Reviewer 3: ```suggestion
- Searchable snapshots are not supported inside shallow copy snapshots.
```
